### ISSUE TITLE
[Bug]: pimcore_block().getCurrent() gives wrong index, when adding an element

### ISSUE THREAD

**Issue by HaaseIT** [pimcore/pimcore#9936]
### Expected behavior

Returns the current index

### Actual behavior

Always returns 0

### Steps to reproduce

This is only when adding the block. If you save the page, the proper index is returned.

```
{% for i in pimcore_iterate_block(pimcore_block('listblock', { "default": 1 })) %}
    {{ pimcore_block('listblock').getCurrent() + 1 }}
{% endfor %}
```

**Comment by dvesh3** [comment]
I'm afraid with current behavior(no reload), it is not possible to generate content structure dynamically. Since, it uses the initially rendered content on adding new block index, you cannot change the structure. 

We have to update the docs to use `reload=true` for these scenarios.

**Comment by HaaseIT** [comment]
This did work properly in 6.x though.

**Comment by HaaseIT** [comment]
Correction: in 6.9 it will reload when a block is added.

**Comment by brusch** [comment]
Fixed by #9938
